source/common/router/req_header_formatter.cc

```diff
@@ -0,0 +1,99 @@
+#include "common/router/req_header_formatter.h"
+
+#include <cstdint>
+#include <iostream>
+#include <map>
+#include <string>
+#include <vector>
+
+#include "common/http/access_log/access_log_formatter.h"
+#include "common/http/headers.h"
+#include "common/json/json_loader.h"
+
+namespace Envoy {
+namespace Router {
+
+HeaderFormatterSharedPtr RequestHeaderParser::parseInternal(const std::string& format) {
+  std::string variable_name;
+  if (format.find("%") == 0) {
+    size_t last_occ_pos = format.rfind("%");
+    if (last_occ_pos == std::string::npos || last_occ_pos <= 1) {
+      throw EnvoyException(fmt::format(
+          "Incorrect configuration: {}. Expected the variable to be of format %<variable_name>%",
+          format));
+    }
+    variable_name = format.substr(1, last_occ_pos - 1);
```

Comment:
Why do we only support a single variable? I had in mind that this could be equally powerful to the access log formatter (and reuse its parsing code).